src/request/RequestHandler.ts

```diff
@@ -19,9 +24,73 @@ export class RequestHandler {
   public static async createRequestObjectStub(
     connectionPolicy: ConnectionPolicy,
     requestOptions: RequestOptions,
-    body: Buffer
+    body?: any
   ) {
-    return createRequestObject(connectionPolicy, requestOptions, body);
+    let didTimeout: boolean;
+    const controller = new AbortController();
+    const signal = controller.signal;
+    const timeout = setTimeout(() => {
+      didTimeout = true;
+      controller.abort();
+    }, connectionPolicy.RequestTimeout);
+
+    let response: any;
+
+    try {
+      // TODO Remove any
+      response = await fetch((requestOptions as any).href + requestOptions.path, {
+        method: requestOptions.method,
+        headers: requestOptions.headers as any,
+        agent: requestOptions.agent,
+        signal,
+        ...(body && { body })
+      } as any); // TODO Remove any. Upstream issue https://github.com/lquixada/cross-fetch/issues/42
+    } catch (error) {
+      if (error.name === "AbortError") {
+        if (didTimeout === true) {
+          throw new TimeoutError();
```

Comment:
We currently expect this to throw ECONNRESET when we successfully hungup, then we try again, so we need to figure out how to handle retries. I need to look into how cross-fetch works in this case.